### PR TITLE
Specviz gui tests

### PR DESCRIPTION
**Pull request by javerbukh**
specviz_gui is created in conftest so that we can write tests that interact with the gui. Just use specviz_gui as an argument in a test method and you will have access to the gui.

### CODE REVIEW

File: specviz/data/ui/plot_window.ui [hunk 1]
```diff
@@ -71,6 +71,7 @@
    <addaction name="remove_region_action"/>
    <addaction name="plot_options_action"/>
    <addaction name="export_plot_action"/>
+   <addaction name="change_unit_action"/>
```
Comment (nmearl) [inline specviz/data/ui/plot_window.ui]: Are these unit dialog ui changes different from the ones in your other PR?

File: specviz/widgets/plotting.py [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
-                            QToolButton, QSizePolicy, QWidget, QMenu, QWidgetAction, QColorDialog, QMessageBox)
+                            QToolButton, QSizePolicy, QWidget, QMenu, QWidgetAction, QColorDialog, QMessageBox, QDialogButtonBox)
 from qtpy.uic import loadUi
 
 from ..core.items import PlotDataItem
 from ..core.models import PlotProxyModel
 from ..utils import UI_PATH
 from .custom import LinearRegionItem
 
+logging.basicConfig(level=logging.DEBUG, format="%(filename)s: %(levelname)8s %(message)s")
```
Comment (nmearl) [inline specviz/widgets/plotting.py]: Are these different than those in #429?

File: specviz/data/ui/plot_window.ui (hunk 1 above)
Comment (javerbukh) [reply specviz/data/ui/plot_window.ui]: Nope

File: specviz/widgets/plotting.py (hunk 2 above)
Comment (javerbukh) [reply specviz/widgets/plotting.py]: No, I'm removing them now

Comment (javerbukh) [reply specviz/widgets/plotting.py]: I've noticed you used logging in the setter functions for data_unit and spectral_axis, should I just leave it in?

**Comment by nmearl** [comment]
Could you also remove the `.pytest_cache` files here too?

**Comment by javerbukh** [comment]
@nmearl They are removed

**Comment by javerbukh** [comment]
I'm not sure how, I haven't merged or made any changes to this since I reverted the helpers last time

**Comment by javerbukh** [comment]
Made outdated by PR #489

**Pull request closed without merging**